benchmarks/benchmarks/optimize_linprog.py

```diff
@@ -274,3 +282,41 @@ def setup(self, meth, prob):
 
     def time_netlib_presolve(self, meth, prob):
         _presolve(self.lp_cleaned, rr=False, tol=1e-9)
+
+
+class Netlib_infeasible(Benchmark):
+    params = [
+        methods,
+        infeasible_problems
+    ]
+    param_names = ['method', 'problems']
+
+    def setup(self, meth, prob):
+        dir_path = os.path.dirname(os.path.realpath(__file__))
+        datafile = os.path.join(dir_path, "linprog_benchmark_files",
+                                "infeasible", prob + ".npz")
+        data = np.load(datafile, allow_pickle=True)
+        self.c = data["c"]
+        self.A_eq = data["A_eq"]
+        self.A_ub = data["A_ub"]
+        self.b_ub = data["b_ub"]
+        self.b_eq = data["b_eq"]
+        self.bounds = np.squeeze(data["bounds"])
+        self.status = None
+
+    def time_netlib_infeasible(self, meth, prob):
+        method, options = meth
+        res = linprog(c=self.c,
+                      A_ub=self.A_ub,
+                      b_ub=self.b_ub,
+                      A_eq=self.A_eq,
+                      b_eq=self.b_eq,
+                      bounds=self.bounds,
+                      method=method,
+                      options=options)
+        self.status = res.status
+
+    def track_netlib_infeasible(self, meth, prob):
```

Comment:
@mdhaber can you explain why this is a benchmark rather than a unit test?

I had to look up `track_*` in the `asv` docs: 
*It is also possible to use asv to track any arbitrary numerical value. “Tracking” benchmarks can be used for this purpose and use the prefix track. These functions simply need to return a numeric value. For example, to track the number of objects known to the garbage collector at a given state:*
So that makes sense, the benchmark is a number that needs to be kept low (memory use, number of objects, etc.). Here however the check is simply whether `res.status` has a numerical value - it needs to be correct rather than low I'd think.

Reply:
Besides being too slow to run with the regular test suite (yes, they could be slow unit tests):
* We _also_ want them to be timed, so I thought it made sense to keep all the code together. (I suppose I could import the code in the unit tests?)
* If they were unit tests, they would not all pass

I suppose the last reason is the most important. These are distributed as benchmarks because they are (or were, at the time) numerically challenging. `linprog` isn't perfect, and we can't fix it overnight, so we want to track its improvements over time.

Reply:
Ah yes that makes sense. If a PR fixes one or more of these problems, they can be moved to the test suite at that point, so they stay fixed.

Reply:
👍 I think that sounds reasonable.

One drawback is that we won't get any CI coverage which would be useful. I don't think any of the benchmark problems are currently tested - as they're too slow - perhaps, in the long run, it is worth adding "formal" tests that are marked as slow/xfail?